src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -154,15 +177,55 @@ public function theCustomerChoseShippingToWithPayment(
         $order->setShippingAddress($address);
         $order->setBillingAddress($address);
 
+        $this->orderRecalculator->recalculate($order);
+
+        $this->objectManager->flush();
+    }
+
+    /**
+     * @Given /^the customer chose ("[^"]+" shipping method) with ("[^"]+" payment)$/
+     */
+    public function theCustomerChoseShippingWithPayment(
+        ShippingMethodInterface $shippingMethod,
+        PaymentMethodInterface $paymentMethod
+    ) {
+        /** @var OrderInterface $order */
+        $order = $this->sharedStorage->get('order');
+
+        $this->orderShipmentFactory->processOrderShipment($order);
+        $order->getShipments()->first()->setMethod($shippingMethod);
+
+        $this->orderRecalculator->recalculate($order);
+
+        $payment = $this->paymentFactory->createWithAmountAndCurrency($order->getTotal(), $order->getCurrency());
+        $payment->setMethod($paymentMethod);
+
+        $order->addPayment($payment);
+
         $this->objectManager->flush();
     }
 
     /**
      * @Given the customer bought a single :product
+     * @Given /^the customer bought ((?:a|an) "[^"]+") and ((?:a|an) "[^"]+")$/
+     */
+    public function theCustomerBoughtProduct(ProductInterface $product, ProductInterface $secondProduct = null)
```

Comment:
It would be more readable to split this function to two different:
```
     /**
       * @Given the customer bought a single :product
      */
     public function theCustomerBoughtProduct(ProductInterface $product)
     {
         $this->addProductVariantToOrder($product->getMasterVariant(), $product->getPrice(), 1);
 
         $this->objectManager->flush();
     }

     /**
      * @Given /^the customer bought ((?:a|an) "[^"]+") and ((?:a|an) "[^"]+")$/
      */
     public function theCustomerBoughtProduct(ProductInterface $product, ProductInterface $secondProduct = null)
     {
         $this->addProductVariantToOrder($product->getMasterVariant(), $product->getPrice(), 1);
         $this->addProductVariantToOrder($secondProduct->getMasterVariant(), $secondProduct->getPrice(), 1);
 
         $this->objectManager->flush();
     }
```
or 
```
     /**
       * @Given the customer bought a single :product
      */
     public function theCustomerBoughtProduct(ProductInterface $product)
     {
         $this->addProductVariantToOrder($product->getMasterVariant(), $product->getPrice(), 1);
 
         $this->objectManager->flush();
     }

     /**
      * @Given /^the customer bought ((?:a|an) "[^"]+") and ((?:a|an) "[^"]+")$/
      */
     public function theCustomerBoughtProduct(ProductInterface $product, ProductInterface $secondProduct = null)
     {
         $this->theCustomerBoughtProduct($product);
         $this->theCustomerBoughtProduct($secondProduct);
 
         $this->objectManager->flush();
     }
```